android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -2163,10 +2148,14 @@ public void setDrmLicenseHeader(String[] header){
     public void setAnalyticsMeta(ReadableMap analyticsData) {
         this.analyticsMeta = analyticsData;
 
-        if (analyticsData != null && analyticsData.getBoolean("contentIsLive") && srcChanged) {
-            if (player != null && analyticsData != null && youboraPlugin == null && contentId != analyticsData.getString("contentId")) {
-                initialiseYoubora();
-            }
+        if (player != null && youboraPlugin == null && (analyticsMeta != null && analyticsMeta.getBoolean("contentIsLive") && contentId != analyticsMeta.getString("contentId"))) {
+            initialiseYoubora();
+        }
+
+        if(player != null && analyticsData == null && youboraPlugin != null && youboraPlugin.getAdapter() != null) {
```

Comment:
This is a bit unreadable, partly because you're duplicating many of the checks above - you can put those in a single if block in the outside, but also because I think it should be an if...else (you never want to both initialise youbora and shut it down).

Reply:
updated using if else to separate init and stop, also add comment

Reply:
Still doesn't really make sense to me. The outer if I was suggesting was for the duplicated `player != null` check, not for analyticsData.

Reply:
I will update this part